tests/integration/test_interface.py

```diff
@@ -300,6 +302,164 @@ def test_dataset_filter_by_year_with_full_dates(mocker, dataset_name: str):
         assert (years == year).all()
 
 
+@pytest.mark.parametrize(
+    "dataset_name",
+    [
+        DATASETS.census.name,
+        DATASETS.acs.name,
+        DATASETS.cps.name,
+        DATASETS.tax_w2_1099.name,
+        DATASETS.wic.name,
+        "TODO: tax_1040",
+    ],
+)
+def test_generate_dataset_with_state_noised(dataset_name: str, request, tmpdir):
+    """Test that dataset generation filtered by state noises data differently with different seeds"""
+    if "TODO" in dataset_name:
+        pytest.skip(reason=dataset_name)
+    data = request.getfixturevalue(f"sample_data_{dataset_name}_state_edit")
+    _generate_non_sample_data_root(dataset_name, tmpdir, data)
+
+    noising_function = DATASET_GENERATION_FUNCS[dataset_name]
+    noised_data = noising_function(source=tmpdir, seed=0, state=STATE)
```

Comment:
Nit: Many of the tests in this file are mocking out the actual noising. Should they actually be considered unit tests rather than integration tests?

Reply:
I wanted to test the filters work in conjunction because they are eventually melded together in the implementation. It seemed like a future change could break that.

I think the tests mocking noising here are integration tests. We're running the full functionality, intentionally avoiding noising so we can test filtering. We can't test filtering is working with noising, since dates and states can be noised. In the future, if/when we have an option to skip noising, we could leverage that.